src/runtime/virtcontainers/nydus.go

```diff
@@ -0,0 +1,349 @@
+package virtcontainers
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/kata-containers/kata-containers/src/runtime/pkg/katautils/katatrace"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	infoEndpoint  = "/api/v1/daemon"
+	mountEndpoint = "/api/v1/mount"
+
+	defaultHttpClientTimeout = 30 * time.Second
+	contentType              = "application/json"
+
+	nydusRafsType          = "rafs"
+	nydusPassthroughfsType = "passthrough_fs"
+
+	configPrefix   = "config="
+	snapshotPrefix = "snapshotdir="
+
+	sharedPathInGuest = "/containers"
+)
+
+var nydusdTracingTags = map[string]string{
+	"source":    "runtime",
+	"package":   "virtcontainers",
+	"subsystem": "nydusd",
+}
+
+type nydusd struct {
+	pid         int
+	path        string
+	sockPath    string
+	apiSockPath string
+	sourcePath  string
+	logFile     string
+	extraArgs   []string
+}
+
+func (nd *nydusd) Start(ctx context.Context, onQuit onQuitFunc) (int, error) {
+	span, _ := katatrace.Trace(ctx, nd.Logger(), "Start", nydusdTracingTags)
+	defer span.End()
+	pid := 0
+
+	args := []string{
+		"--log-level", "debug",
+		"--log-file", nd.logFile,
```

Comment:
nydusd output is already caught below, why is the logfile still needed?

Reply:
I have removed --log-file